toast-tk-web-agent/src/main/java/io/toast/tk/agent/ui/ConfigPanel.java

```diff
@@ -412,215 +173,156 @@ private void close() {
-		{
-			if(!testWebAppURL(textFieldRecorder.getText(),runTryValue))
-			{
-				iconPanelRecorder.removeAll();
-				iconPanelRecorder.add(iconNotValidRecorder);
-				errorLabelRecorder.setText(errorMessageSelectURL);
-			}
-			else 
-			{
-				iconPanelRecorder.removeAll();
-				iconPanelRecorder.add(iconValidRecorder);
-				errorLabelRecorder.setText(" ");
-			}
-		}
-		if(strKey.contains(chromeDriverName)) {
-			if(!testWebAppDirectory(textFieldChrome.getText(),runTryValue))
-			{
-				iconPanelChrome.removeAll();
-				iconPanelChrome.add(iconNotValidChrome);
-				errorLabelChrome.setText(errorMessageSelectFile);
-			}
-			else 
-			{
-				iconPanelChrome.removeAll();
-				iconPanelChrome.add(iconValidChrome);
-				errorLabelChrome.setText(" ");
-			}
-		}
+	private JPanel buildProxyPanel() {
+		JPanel proxyPanel = PanelHelper.createBasicPanel(BoxLayout.PAGE_AXIS);
+	    proxyPanel.add(proxyCheckBox);
+	    proxyPanel.add(proxyAdressPanel);
+	    proxyPanel.add(proxyPortPanel);
+	    proxyPanel.add(proxyUserNamePanel);
+	    proxyPanel.add(proxyUserPswdPanel);
+	    return proxyPanel;
 	}
+
+	private JPanel buildFullButtonPanel() {
+		JPanel buttonPanel = PanelHelper.createBasicPanel();
+		buttonPanel.setAlignmentX(LEFT_ALIGNMENT);
+		buttonPanel.setLayout(new BoxLayout(buttonPanel, BoxLayout.LINE_AXIS));
+		buttonPanel.add(Box.createHorizontalGlue());
+		buttonPanel.setBorder(BorderFactory.createEmptyBorder(15, 0, 0, 0));
+		buttonPanel.setBackground(Color.white);
+		buttonPanel.add(createBasicTestButton());
+		buttonPanel.add(createBasicCancelButton());
+		buttonPanel.add(createBasicOkButton());
+		return buttonPanel;
+	}
+	
+	private JButton createBasicOkButton() {
+		JButton okButton = new JButton("Ok");
 		
-	public static boolean testWebAppDirectory(String directory, boolean runTryValue) throws IOException{
-		String fileName = directory.split("/")[directory.split("/").length - 1];
-		
-		if(directory.contains(" ")){
-			if(runTryValue) 
-    		{
-				NotificationManager.showMessage(directory + " has spaces in its name.").showNotification();
-    		}
-			LOG.info("Status of " + directory + " : KO");
-    		return false;
-    	}
-    	
-    	if(testFileDirectory(directory, runTryValue, fileName)){
-    		LOG.info("Status of " + directory + " : OK");
-    		return true;
-    	}
-    	else {
-    		
-    		LOG.info("Status of " + directory + " : KO");
-    		return false;
-    	}
+		okButton.addActionListener(new ActionListener() {
+			@Override
+			public void actionPerformed(ActionEvent e) {
+				saveAndClose();				
+			}
+		});
+		return okButton;
 	}
-
-	public static boolean testFileDirectory(String directory, boolean runTryValue, String fileName) {
-		File myFile = new File(directory); 
-		if(myFile.exists()) 
-		{
-			if(myFile.isFile())
-			{
-				return true;
+	
+	private JButton createBasicCancelButton() {
+		JButton cancelButton = new JButton("Cancel");
+		cancelButton.addActionListener(new ActionListener() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Make this anonymous inner class a lambda [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1604)